Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3653,7 +3669,23 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
 
                         // Output content is reported here to ensure that it happens both on worker executing PostProcess and
                         // master which called worker to execute post process.
-                        PipExecutor.ReportExecutionResultOutputContent(operationContext, environment, processRunnable.Description, executionResult, processRunnable.Process.DoubleWritePolicy.ImpliesDoubleWriteIsWarning());
+                        PipExecutor.ReportExecutionResultOutputContent(
+                            operationContext,
+                            environment,
+                            processRunnable.Description,
+                            executionResult,
+                            processRunnable.Process.DoubleWritePolicy.ImpliesDoubleWriteIsWarning());
+
+                        PipExecutor.ReportSourceChangeAffectedOutputs(

```

Comment:
>PipExecutor.ReportSourceChangeAffectedOutputs( [](start = 24, length = 46)

Does the worker need to know about the affected outputs? Or should the master send that information to the worker?

Reply:
The global affected outputs are only maintained in master. When master distribute the runnerablepip, the change affected inputs of the pip was computed in the master and set in the runnerablepip. The worker will write out this affected inputs to a file.